### ISSUE TITLE
A file matching same participants over different submission IDs

### ISSUE THREAD

**Issue by renelabounek** [sct-pipeline/gm-challenge-data#7]
sub-10062 and sub-10063 is the same participant.

If another centers have also submitted multiple datasets, there might be a similar issue for another IDs.

**Comment by jcohenadad** [comment]
thank you for bringing that up. @alexfoias is there a BIDS friendly way to solve that problem?

**Comment by renelabounek** [comment]
I do not think that remove is what we wanted as 10062 and 10063 are two parallel protocols tested on the same subject.

I believe similar thing applies for 9709-9710 and 9611-9612-9613

**Comment by alexfoias** [comment]
Sorry, I have misunderstood. I will close the PR.

**Comment by jcohenadad** [comment]
@alexfoias can you please discuss your strategy here and wait for us to validate before implementing it #18 #17

**Comment by alexfoias** [comment]
The proposed strategy in #18 #17 was to we have to mix (under ses-XX) :

- 10062-10063
- 9709-9710
- 9611-9612-9613

If that doesn't work, we can mention this thing in `README` or by adding another column in `participants.tsv` to specify this information.

**Comment by jcohenadad** [comment]
> by adding another column in `participants.tsv` to specify this information.

can you please elaborate?

**Comment by alexfoias** [comment]
What about adding a column `same_sub_submission` ?

`participants.tsv` would look like:

|participant_id|sex|age|same_sub_submission|
|--------------|---|---|-------------------|
|sub-9418|M|31|unique|
|sub-9584|M|27|unique|
|sub-9604|M|n/a|unique|
|sub-9605|M|37|unique|
|sub-9611|n/a|n/a|sub-9611|
|sub-9612|n/a|n/a|sub-9611|
|sub-9613|n/a|n/a|sub-9611|
|sub-9669|n/a|n/a|unique|
|sub-9709|F|35|sub-9709|
|sub-9710|F|35|sub-9709|
|sub-10062|n/a|n/a|sub-10062|
|sub-10063|n/a|n/a|sub-10062|
|sub-10328|n/a|n/a|unique|
|sub-20210728|F|24|unique|

**Comment by jcohenadad** [comment]
I find the approach of https://github.com/sct-pipeline/gm-challenge-data/issues/7#issuecomment-897784736 a little hard to grasp. How about simply introducing a column `participant_id_unique` with the following fields:

|participant_id|sex|age|participant_id_unique|
|--------------|---|---|-------------------|
|sub-9418|M|31|9418|
|sub-9584|M|27|9584|
|sub-9604|M|n/a|9604|
|sub-9605|M|37|9605|
|sub-9611|n/a|n/a|9611|
|sub-9612|n/a|n/a|9611|
|sub-9613|n/a|n/a|9611|
|sub-9669|n/a|n/a|9669|
|sub-9709|F|35|9709|
|sub-9710|F|35|9709|
|sub-10062|n/a|n/a|10062|
|sub-10063|n/a|n/a|10062|
|sub-10328|n/a|n/a|10328|
|sub-20210728|F|24|20210728|

That way we can easily count the number of unique subjects algorithmically.

**Comment by renelabounek** [comment]
I think the participant_id_unique suggestion with the first smallest ID sounds as a quite good deal. Another way, how it might be solved is change particpant_id, as e.g.:

sub-9611-1
sub-9611-2
sub-9611-3

It would reduce the file participatns.tsv back to 2 columns. And regarding bids organization of potential longitudinal studies, it would make me more sense to organize datasets in foldernames which consist both the participant ID and the session ID. This way is also easy to algoritmize and idnetify unique subjects. I am using it. People at the CMRR usually use it too.

I confirm that Oxford submission (9611-9613) is the same subject. Johanna Vannesjo's comment in the manuscript:
_"True for Oxford - 9611-9613 is the exact same acquisition, but with different combinations of the images from different echo times."_
I have not seen any confirmation message from Milan.

Another way (which I like less) how to change the dataset tree can be:

sub-9611/1/anat/*
sub-9611/1/dwi/*
sub-9611/2/anat/*
sub-9611/2/dwi/*
sub-9611/3/anat/*
sub-9611/3/dwi/*

Lots of people use this kind of structure too. But I am not sure if bids is fan of this strategy, or is not. I prefer my first suggestion, if there should be a change compared to the PR #20 suggestion.

**Comment by renelabounek** [comment]
I am not sure, if I was clear file participants.tsv would have single sow sub-9611 and database would have 3 folders (sub-9611-1, sub-9611-2, sub-9611-3) in the root folder.

I see dcm2bids has a flag -s for session ID. Then, it should not be hard to convert data into bids format as multiple sessions for same subject ID.

**Comment by renelabounek** [comment]
When you call dcm2bids as: `dcm2bids -p sub-0002 -s 01` the output folder tree appears like:
sub-0002/ses-01/anat/*
sub-0002/ses-01/dwi/*
So dcm2bids works more like the 2nd suggestion which I have written. Would it be hard to rewrite gmchallenge analysis into this folder structure?

**Comment by jcohenadad** [comment]
i definitely like @renelabounek's idea better:
~~~
sub-9611-1
sub-9611-2
sub-9611-3
~~~

@alexfoias can you please check if it would pass BIDS compliance?

**Comment by alexfoias** [comment]
I tried it and it doesn't like it. 

Here is the output of a test:

```
bids-validator@1.8.2

	1: [ERR] Files with such naming scheme are not part of BIDS specification. This error is most commonly caused by typos in file names that make them not BIDS compatible. Please consult the specification and make sure your files are named correctly. If this is not a file naming issue (for example when including files not yet covered by the BIDS specification) you should include a ".bidsignore" file in your dataset (see https://github.com/bids-standard/bids-validator#bidsignore for details). Please note that derived (processed) data should be placed in /derivatives folder and source data (such as DICOMS or behavioural logs in proprietary formats) should be placed in the /sourcedata folder. (code: 1 - NOT_INCLUDED)
		./README.md
			Evidence: README.md
		./sub-9418-1/anat/sub-9418-1_run-1_T2starw.json
			Evidence: sub-9418-1_run-1_T2starw.json
		./sub-9418-1/anat/sub-9418-1_run-1_T2starw.nii.gz
			Evidence: sub-9418-1_run-1_T2starw.nii.gz

	Please visit https://neurostars.org/search?q=NOT_INCLUDED for existing conversations about this issue.

	2: [ERR] Sub label contain an Illegal Character hyphen or underscore. Please edit the filename as per BIDS spec. (code: 62 - SUBJECT_VALUE_CONTAINS_ILLEGAL_CHARACTER)
		./sub-9418-1/anat/sub-9418-1_run-1_T2starw.json
			Evidence: sub name contains illegal character:/sub-9418-1/anat/sub-9418-1_run-1_T2starw.json

	Please visit https://neurostars.org/search?q=SUBJECT_VALUE_CONTAINS_ILLEGAL_CHARACTER for existing conversations about this issue.

	3: [ERR] No BIDS compatible data found for at least one subject. (code: 67 - NO_VALID_DATA_FOUND_FOR_SUBJECT)
		./sub-9418-1

	Please visit https://neurostars.org/search?q=NO_VALID_DATA_FOUND_FOR_SUBJECT for existing conversations about this issue.

	1: [WARN] The recommended file /README is missing. See Section 03 (Modality agnostic files) of the BIDS specification. (code: 101 - README_FILE_MISSING)

	Please visit https://neurostars.org/search?q=README_FILE_MISSING for existing conversations about this issue.

	2: [WARN] The Authors field of dataset_description.json should contain an array of fields - with one author per field. This was triggered because there are no authors, which will make DOI registration from dataset metadata impossible. (code: 113 - NO_AUTHORS)

	Please visit https://neurostars.org/search?q=NO_AUTHORS for existing conversations about this issue.


        Summary:               Available Tasks:        Available Modalities: 
        7 Files, 1.96MB                                                      
        0 - Subjects                                                         
        1 - Session                                                          


	If you have any questions, please post on https://neurostars.org/tags/bids.


```

**Comment by jcohenadad** [comment]
how about
~~~
sub-9611Ses1
sub-9611Ses2
sub-9611Ses3
~~~

**Comment by alexfoias** [comment]
That one works.

**Comment by jcohenadad** [comment]
let's go for it then
